### PR TITLE
Alternative fix to #2643: allow autowiring of filter classes

### PR DESCRIPTION
**Pull request by meyerbaptiste**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/api-platform/core/pull/2643#issuecomment-476202544
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/ApiPlatformBundle.php [hunk 1]
```diff
@@ -36,7 +37,7 @@ public function build(ContainerBuilder $container)
         parent::build($container);
 
         $container->addCompilerPass(new DataProviderPass());
-        $container->addCompilerPass(new AnnotationFilterPass());
+        $container->addCompilerPass(new AnnotationFilterPass(), PassConfig::TYPE_BEFORE_OPTIMIZATION, 101);
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/ApiPlatformBundle.php]: We need to document why 101 :smile:

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 2]
```diff
@@ -16,10 +16,12 @@
 use ApiPlatform\Core\Api\FilterInterface;
 use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Extension\AggregationCollectionExtensionInterface;
 use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Extension\AggregationItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Filter\AbstractFilter as DoctrineMongoDbOdmAbstractFilter;
 use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\EagerLoadingExtension;
 use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\FilterEagerLoadingExtension;
 use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryCollectionExtensionInterface as DoctrineQueryCollectionExtensionInterface;
 use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\AbstractContextAwareFilter as DoctrineOrmAbstractFilter;
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: Probably clearer as `DoctrineOrmAbstractContextAwareFilter`?

File: src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/AnnotationFilterPass.php [hunk 3]
```diff
@@ -13,15 +13,14 @@
 
 namespace ApiPlatform\Core\Bridge\Symfony\Bundle\DependencyInjection\Compiler;
 
-use ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm\Filter\FilterInterface as MongoDbOdmFilterInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/AnnotationFilterPass.php]: Should probably not throw our exception here, but something from https://github.com/symfony/symfony/tree/master/src/Symfony/Component/DependencyInjection/Exception

**Comment by meyerbaptiste** [comment]
Done! 😃

**Comment by teohhanhui** [comment]
Thanks @meyerbaptiste! :tada: :rocket:

**Pull request merged**